pandas/indexes/base.py

```diff
@@ -1745,29 +1745,13 @@ def argsort(self, *args, **kwargs):
         return result.argsort(*args, **kwargs)
 
     def __add__(self, other):
-        if is_list_like(other):
-            warnings.warn("using '+' to provide set union with Indexes is "
-                          "deprecated, use '|' or .union()", FutureWarning,
-                          stacklevel=2)
-        if isinstance(other, Index):
```

Comment:
Yes, for MultiIndex/Categorical nothing should be changed. I just renamed the method to disable add/sub for those, as there are no longer set ops.

For `Index`, I agree we should be more careful on what to allow, but I think that this is already the case in current master. 
The 'old' `__add__` did basically:

```
def __add__(self, other):
    if isinstance(other, Index):
        self.union(other)
    return Index(self.values + other)
```

So by removing the `if instance` part I am not broading the scope, previously everything that was not an Index was already handled to do addition. 

But, as I said, I agree we should be more careful / do more careful testing (eg Int64Index does not allow to add with lists, base Index does). 
I will see if I do that here, or leave it for another issue.